Userland/Libraries/LibWeb/DOM/Node.cpp

```diff
@@ -606,4 +606,34 @@ RefPtr<Document> Node::owner_document() const
     return m_document;
 }
 
+void Node::dump_tree_json(JsonObjectSerializer<StringBuilder>& object) const
+{
+    object.add("name", node_name().view());
+    object.add("internal_id", (size_t)this);
+    if (is_document())
+        object.add("type", "document");
+    else if (is_element())
+        object.add("type", "element");
+    else if (is_text())
+        object.add("type", "text");
+
+    if (is_element()) {
+        auto element = downcast<DOM::Element>(this);
+        if (element->has_attributes()) {
+            auto attributes = object.add_object("attributes");
+            element->for_each_attribute([&attributes](auto& name, auto& value) {
+                attributes.add(name, value);
+            });
+        }
+    }
+
```

Comment:
You could check `is_text()` here, and if that is the case, you could:
```cpp
else if (is_text()) {
    auto text_node = downcast<DOM::Text>(this);
    object.add("text", text_node->data());
}
```
This way, you could get the text contents into the JSON tree.

Reply:
I was unsure if this text would cause performance issues, but I think the performance impact is elsewhere, so I will add this :+1: 

I also noticed I'm doing `is_element()` twice, so can simplify that :)